README


This archive contains the distribution Dist-Zilla-PluginBundle-Author-MELO,
version 0.011:

  MELO is lazy, this are his rules

This software is Copyright (c) 2011 by Pedro Melo.

This is free software, licensed under:

  The Artistic License 2.0 (GPL Compatible)




Changes
Revision history for Dist-Zilla-PluginBundle-Author-MELO

{{$NEXT}}

0.012     2013-01-18T11:29:04Z
        * New default for skip_prereqs: '^t::'
        * Tweak SUPPORT section of our Pod sections
        * Switch part of our Metadata collection to the GitHub PluginBundle
        * bug tracker set to Github issues by default now

0.011     2013-01-06T15:49:13Z
        * Generate a simpler .travis.yml
        * Make sure .travis.yml is not excluded by MANIFEST.SKIP
        * Add support for DZIL_FIRST_RELEASE ENV
        * Test::Pod::LinkCheck is disable by default for now
        * Test::Pod::No404s is disable by default for now

0.010     2012-09-05T17:37:47Z
        * Updated abstract to the real truth about this module
        * Generate a .travis.yml file for all projects using this bundle

0.009     2012-09-05T00:36:10Z
        * Fix bug with plugin order: Git::CommitBuild must be
          before @Git

0.008     2012-09-05T00:31:36Z
        * Will the real Git::CommitBuild config work this time?

0.007     2012-09-05T00:15:16Z
        * Fix Git::CommitBuild setup
        * Didn't make it to CPAN, botched up release phase

0.006     2012-09-04T23:59:04Z
        * New release to fix timestamp problems on the previous dist
        * Add Git::CommitBuild with both per-build and per-release
          extra branches
        * Add PrereqsClean plugin
        * Tweak cover_db/ exception

0.005     2012-09-04T11:19:08Z
        * disable_tests accepts multiple values, in line with
          @TestingMania behaviour

0.004     2012-05-11T11:25:40Z
        * Added missing dependency not detected auto-magically

0.003     2012-03-14T10:54:01Z
        * Add default authority to make Dist::Zilla::Plugin::Authority
          work with default travis-ci.org install

0.002     2011-10-28T11:15:39Z
        * Make ReportVersions::Tiny plugin optional and off by default
        * Cleanup after a release

0.001     2011-10-28T10:58:32Z
        * Initial release


lib/Pod/Weaver/PluginBundle/Author/MELO.pm
package Pod::Weaver::PluginBundle::Author::MELO;

BEGIN {

  # VERSION
  # AUTHORITY
}

# ABSTRACT: MELO's Pod::Weaver config

use strict;
use warnings;
use Pod::Weaver 3.101633 ();
use Pod::Weaver::PluginBundle::Default ();
use Pod::Weaver::Plugin::StopWords 1.001005 ();
use Pod::Weaver::Plugin::Transformer ();
## TODO: do we really want this WikiDoc?
use Pod::Weaver::Plugin::WikiDoc 0.093002 ();
use Pod::Weaver::Section::Support 1.001   ();
use Pod::Elemental 0.102360               ();
use Pod::Elemental::Transformer::List ();

use Pod::Weaver::Config::Assembler;
sub _exp { Pod::Weaver::Config::Assembler->expand_package($_[0]) }

sub _plain {
  my ($plug, $arg) = (@_, {});
  (my $name = $plug) =~ s/^\W//;
  return [$name, _exp($plug), {%$arg}];
}

sub _bundle_name {
  my $class = @_ ? ref $_[0] || $_[0] : __PACKAGE__;
  join('', '@', ($class =~ /^.+::PluginBundle::(.+)$/));
}

sub mvp_bundle_config {
  ## ($self, $bundle) = @_; $bundle => {payload => {}, name => '@...'}
  my ($self) = @_;
  my @plugins;

  # NOTE: bundle name gets prepended to each plugin name at the end

  push @plugins, (

    # plugin
    _plain('-Encoding'),
    _plain('-WikiDoc'),

    # default
    _plain('@CorePrep'),

    # sections
    # default
    _plain('Name'),
    _plain('Version'),

    # Any pod inside a =begin/end :prelude will go at the top
    ['Prelude', _exp('Region'), {region_name => 'prelude'}],
  );

  push @plugins, map { $_->[2]{header} = uc($_->[0]); $_ } (

    # default
    ['Synopsis',    _exp('Generic'), {}],
    ['Description', _exp('Generic'), {}],
    ['Overview',    _exp('Generic'), {}],

    # extra
    ['Usage', _exp('Generic'), {}],

    # default
    ['Attributes',   _exp('Collect'), {command => 'attr'}],
    ['Constructors', _exp('Collect'), {command => 'constructor'}],
    ['Methods',      _exp('Collect'), {command => 'method'}],
    ['Functions',    _exp('Collect'), {command => 'func'}],
  );

  # default
  push @plugins, (
    _plain('Leftovers'),

    # see prelude above
    ['Postlude', _exp('Region'), {region_name => 'postlude'}],

    # include Support section with various cpan links and github repo
    [ 'Support',
      _exp('Support'),
      { email => 'melo',

        repository_link    => 'both',

        bugs => 'metadata',
        bugs_content => 'Please report any bugs or feature requests through the web interface at {WEB}. You will be automatically notified of any progress on the request by the system.',

        websites => [qw(metacpan testers testmatrix deps ratings)],
      }
    ],

    ['Acknowledgements', _exp('Generic'), {header => 'ACKNOWLEDGEMENTS'}],

    # default
    _plain('Authors'),
    _plain('Legal'),

    # plugins
    ['List', _exp('-Transformer'), {'transformer' => 'List'}],

    # my dictionary doesn't like that extra 'E' but it looks funny without it
    _plain('-StopWords', {include => 'ACKNOWLEDGEMENTS'}),
  );

  # prepend bundle name to each plugin name
  my $name = $self->_bundle_name;
  @plugins = map { $_->[0] = "$name/$_->[0]"; $_ } @plugins;

  return @plugins;
}

1;


__END__

=encoding utf-8

=head1 NAME

Pod::Weaver::PluginBundle::Author::MELO - MELO's Pod::Weaver config

=head1 SYNOPSIS

  # weaver.ini
  [@Author::MELO]

or with a F<dist.ini> like so:

  # dist.ini
  [@Author::MELO]

you don't need a F<weaver.ini> at all.

=head1 DESCRIPTION

Another fork of the excellent work by RWSTASUNER.

This PluginBundle is like the @Default
with the following additions:

=begin :list

* Inserts a SUPPORT section to the POD just before AUTHOR
* Adds the List Transformer
* Enables WikiDoc formatting
* Generates and collects stopwords

=end :list

It is roughly equivalent to:

  [Encoding]                ; prepend '=encoding utf-8' automatically
  [WikiDoc]                 ; transform wikidoc sections to POD
  [@CorePrep]               ; [@Default]

  [Name]                    ; [@Default]
  [Version]                 ; [@Default]

  [Region  / prelude]       ; [@Default]

  [Generic / SYNOPSIS]      ; [@Default]
  [Generic / DESCRIPTION]   ; [@Default]
  [Generic / OVERVIEW]      ; [@Default]
  [Generic / USAGE]         ; Put USAGE section near the top

  [Collect / ATTRIBUTES]    ; [@Default]
  command = attr

  [Collect / CONSTRUCTORS]
  command = constructor

  [Collect / METHODS]       ; [@Default]
  command = method

  [Collect / FUNCTIONS]     ; [@Default]
  command = func

  [Leftovers]               ; [@Default]

  [Region  / postlude]      ; [@Default]

  ; custom section
  [Support]                 ; =head1 SUPPORT (bugs, cpants, git...)
  repository_content =
  repository_link = both
  websites = testers, testmatrix, deps, ratings

  [Authors]                 ; [@Default]
  [Legal]                   ; [@Default]

  [-Transformer]            ; enable =for :list
  transformer = List

  [-StopWords]              ; generate some stopwords and gather them together

=for Pod::Coverage mvp_bundle_config

=cut


Makefile.PL

use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "MELO is lazy, this are his rules",
  "AUTHOR" => "Pedro Melo <melo\@simplicidade.org>",
  "BUILD_REQUIRES" => {
    "File::Temp" => 0,
    "Test::More" => "0.98"
  },
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-Author-MELO",
  "EXE_FILES" => [],
  "LICENSE" => "artistic_2",
  "NAME" => "Dist::Zilla::PluginBundle::Author::MELO",
  "PREREQ_PM" => {
    "Dist::Zilla" => "4.300002",
    "Dist::Zilla::Plugin::Authority" => "1.005",
    "Dist::Zilla::Plugin::Bootstrap::lib" => "0.01023600",
    "Dist::Zilla::Plugin::Bugtracker" => "1.111080",
    "Dist::Zilla::Plugin::CheckChangesHasContent" => "0.003",
    "Dist::Zilla::Plugin::CheckExtraTests" => "0.004",
    "Dist::Zilla::Plugin::Clean" => "0.02",
    "Dist::Zilla::Plugin::Git::NextVersion" => 0,
    "Dist::Zilla::Plugin::InstallRelease" => "0.007",
    "Dist::Zilla::Plugin::MetaProvides::Package" => "1.12060501",
    "Dist::Zilla::Plugin::MinimumPerl" => "1.003",
    "Dist::Zilla::Plugin::OurPkgVersion" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::PrereqsClean" => 0,
    "Dist::Zilla::Plugin::ReportVersions::Tiny" => "1.03",
    "Dist::Zilla::Plugin::Repository" => "0.18",
    "Dist::Zilla::Plugin::Test::Pod::No404s" => "1.001",
    "Dist::Zilla::PluginBundle::Git" => "1.112510",
    "Dist::Zilla::PluginBundle::TestingMania" => "0.014",
    "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
    "List::Util" => 0,
    "Method::Signatures" => 20111020,
    "Moose" => "1.01",
    "Pod::Elemental" => "0.102360",
    "Pod::Elemental::Transformer::List" => 0,
    "Pod::Weaver" => "3.101633",
    "Pod::Weaver::Plugin::Encoding" => 0,
    "Pod::Weaver::Plugin::StopWords" => "1.001005",
    "Pod::Weaver::Plugin::WikiDoc" => "0.093002",
    "Pod::Weaver::Section::Support" => "1.001",
    "Test::CPAN::Meta::JSON" => 0
  },
  "VERSION" => "0.012",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



